src/panel_gwalker/_gwalker.py

```diff
@@ -164,55 +430,177 @@ def _compute(self, payload):
+           Whether to export and save the chart specification(s) or SVG. If None the mode is set
+           to the 'export_code' parameter value.
+        scope: 'current' | 'all'
+           Whether to export and save only the current chart or all charts. If None the scope is
+           set to the 'export_scope' parameter value.
+        timeout: int
+           How long to wait for the response before timing out. If None the timeout is
+           set to the 'export_timeout' parameter value.
+        """
+        if not path:
+            path = self.save_path
+        spec = await self._export(mode=mode, scope=scope, timeout=timeout)
+        path = Path(path)
+        with path.open("w") as file:
+            json.dump(spec, file)
+        logger.debug("Saved spec to %s", path)
+
+    def create_save_button(self, **params) -> SaveButton:
+        """Returns a UI component to save the chart(s) as either a spec or SVG.
+
+        >>> walker.create_save_button(width=400)
+
+        The spec or SVG will be saved to the path give by `save_path`.
+        """
+        return SaveButton(self, **params)
+
+    def chart(self, index: int | list | None = None, **params) -> "GraphicWalker":
+        """Returns a clone with `renderer='chart'` and `server_computation=False`.
+
+        >>> walker.chart(1, width=400)
+        """
+        params["index"] = index
+        params["renderer"] = "chart"
+        params["server_computation"] = False
+        return self.clone(**params)
+
+    def explorer(self, **params) -> "GraphicWalker":
+        """Returns a clone with `renderer='explorer'`.
+
+        >>> walker.explorer(width=400)
+        """
+        params["renderer"] = "explorer"
+        return self.clone(**params)
+
+    def profiler(self, **params) -> "GraphicWalker":
+        """Returns a clone with `renderer='profiler'`.
+
+        >>> walker.profiler(page_size=50, width=400)
+        """
+        params["renderer"] = "profiler"
+        return self.clone(**params)
+
+    def viewer(self, **params) -> "GraphicWalker":
+        """Returns a clone with `renderer='viewer'`.
+
+        >>> walker.viewer(width=400)
+        """
+        params["renderer"] = "viewer"
+        return self.clone(**params)
+
+    _PARAMETER_IS_ENABLED = {
+        "page_size": ["profiler"],
+        "index": ["viewer", "chart"],
+        "tab": ["explorer"],
+        "container_height": ["viewer", "chart"],
+    }
+
+    @classmethod
+    def _is_enabled(cls, renderer, parameter) -> bool:
+        """Returns True if the parameter is enabled for the renderer."""
+        if not parameter in cls._PARAMETER_IS_ENABLED:
+            return True
+
+        return renderer in cls._PARAMETER_IS_ENABLED[parameter]
+
+    @classmethod
+    def _is_disabled(cls, renderer, parameter) -> bool:
+        """Returns True if the parameter is disabled for the renderer."""
+        return not cls._is_enabled(renderer, parameter)
+
+    def is_enabled(self, parameter: str):
```

Comment:
I don't understand what these are for tbh.

Reply:
Some parameters only work for specific renderers. Users would loose touch of this.

To help users understand which parameters are relevant for a given renderer I hide widgets in the reference app. Initially that functionality was in the reference app. But it took app a lot of space and could confuse users. Also user (like me) that want to add some of these parameters to their applications would have to re-implement this logic.

So these are utility functions to make it easier to develop apps. And to hide complexity.

I could also use these functions to dynamically set for example `self.param.page_size.disabled=self.is_enabled("page_size")`. But I could not see through whether it could cause problems for users. And setting visible=False is also better than disabled=True for reducing complexity in a UI.